### PR TITLE
first iteration of PR template

### PR DESCRIPTION
**Pull request by barrioskevin**
This is basic template we can use for making pull requests. As @misherwa pointed out design document 2 points out some key things that we should take note of. Pull requests happen to be emphasized heavily. In an attempt to make sure that we are all making good pull requests. I propose a template that can make sure our PRs have structure. We should also be looking into a checklist for reviewing pull requests. If you guys have any suggestions to the template let me know and I can update it. The way it's set up we should also be able to include multiple templates so if we ever want to differentiate pull requests in any kind of way we can make an individual template. I also attached the pull request section highlighted in design doc 2 for more context.
image

### CODE REVIEW

**Comment by misherwa** [comment]
Thanks Kevin for starting on the PR Template.

File: .github/PULL_REQUEST_TEMPLATE/pull_request_template.md [hunk 1]
```diff
@@ -0,0 +1,39 @@
+### Description
```
Comment (misherwa) [inline .github/PULL_REQUEST_TEMPLATE/pull_request_template.md:15]: Hey Kevin, I recommend adding a checklist to the template to make PRs easier to understand. You could put the checklist at the top or somewhere else in the document.

Here's what the code would look like:
q

Here's what it would look like when viewing the template:
### Checklist
- [ ] **Feature**: Introduces new code to implement a feature or features.
- [ ] **Issue fix**: Implements any changes to fix a bug or issue.
- [ ] **Test**: Implements new tests or modifying any existing ones.
- [ ] **Domain**: Effecting either the client-side (frontend) or serverside (backend). f applicable, select from the options below:
    - [ ] Frontend
    - [ ] Backend
    - [ ] Both Frontend and Backend
- [ ] **Documentation**: Updates to documentation, including comments, README files, or relevant documents. If applicable, select from the options below:
    - [ ] Comments
    - [ ] README
    - [ ] Other Documents
- [ ] **Other**: Captures other key changes or details not covered in the checklist options above.



I can update this file and add it myself if you think it's good. Or if you want to copy and paste the code for yourself, here it is:
# ### Checklist
# - [ ] **Feature**: Introduces new code to implement a feature or features.
# - [ ] **Issue fix**: Implements any changes to fix a bug or issue.
# - [ ] **Test**: Implements new tests or modifies any existing ones.
# - [ ] **Domain**: Affects either the client-side (frontend) or server-side (backend). If applicable, select from the options below:
#     - [ ] Frontend
#     - [ ] Backend
#     - [ ] Both Frontend and Backend
# - [ ] **Documentation**: Updates documentation, including comments, README files, or other relevant documents. If applicable, select from the options below:
#     - [ ] Comments
#     - [ ] README
#     - [ ] Other Documents
# - [ ] **Other**: Captures other key changes or details not covered in the checklist options above.

Comment (barrioskevin) [reply .github/PULL_REQUEST_TEMPLATE/pull_request_template.md:15]: yeah i suppose that's a good idea but if we include this we should come up with certain boxes that like need to be checked.

**Comment by barrioskevin** [comment]
I added a simple checklist, I think if we are going to include a checklist that it should include categories with simple labels for any of us to take a quick look at see what it's about. We should make them specific to rebel hubs yk like we can add category for the different areas of app, im not too sure. feel free to make any adjustments

**Comment by misherwa** [comment]
Gotcha.

For me, I feel like the only thing you could change is the Testing and Mentions sections. 

The Testing section has a different tone in the second sentence " The prof wants us to be writing unit tests and I even think he expects us to apply TTD" from the other sections and sentences. I feel like we should keep the writing in our template fairly professional for the template. Also in the same sentence, I think it's supposed to be "TDD" and not "TTD".

I'd recommend changing it to either this first one: 
"The professor expects us to write unit tests, with an emphasis on applying Test-Driven Development (TDD)."

Or this second one:
"We are expected to write unit tests, with a strong emphasis on following Test-Driven Development (TDD)."

I'm feeling the second second more. But for the Mentions section, I would just recommend capitalizing all "pr" to "PR", just so it's the capitalization is consistent throughout the template.

I added some comments to the file you changed so you can get a better look at it right below.

File: .github/PULL_REQUEST_TEMPLATE/pull_request_template.md [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+as developers should be using your code. 
+
+### Testing
+Testing seems to be a very important part of this class. The prof wants us to be writing unit tests and I even think he expects us to
+apply TTD. We should at the very least describe what kind of testing we did to ensure that these new additions work properly. Also list out any
+edge cases you encountered or are suspicious of. Ideally we want to be writing unit tests and be able to provide the coverage here. If you
+have implemented unit test make sure to provide the commands necessary for others to run your tests.
+
+### Mentions
+Tag any group members that you wish to review your pr, helped with your pr, or need to take note of your pr.
```
Comment (misherwa) [inline .github/PULL_REQUEST_TEMPLATE/pull_request_template.md]: Capitalization of "pr" to make the sentence look like:

"Tag any group members that you wish to review your PR, helped with your PR, or need to take note of your PR."

File: .github/PULL_REQUEST_TEMPLATE/pull_request_template.md [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+Here you can go more in depth and explain the underlying architecture of the system that you are introducing. If you are
+making changes to a previous system that just explain any differences. We should be able to understand your though process behind the
+code that you have written. 
+
+### Use of your changes
+Here you can explain how you expect the different systems of our app to interact with your new additions. You can also explain how us 
+as developers should be using your code. 
+
+### Testing
+Testing seems to be a very important part of this class. The prof wants us to be writing unit tests and I even think he expects us to
```
Comment (misherwa) [inline .github/PULL_REQUEST_TEMPLATE/pull_request_template.md]: Changing this sentence in the Testing section "The prof wants us to be writing unit tests and I even think he expects us to apply TTD." 

To either this: "The professor expects us to write unit tests, with an emphasis on applying Test-Driven Development (TDD)."

Or this: "We are expected to write unit tests, with a strong emphasis on following Test-Driven Development (TDD)."

**Comment by barrioskevin** [comment]
Hello @misherwa , I apologize for not keeping the tone as professional. I don't know what I was thinking, I think as i started to write the testing section I started to talk more directly to you guys as a group. I am not the best when it comes to communication so I went ahead and had ChatGPT rewrite the template for me. I like what it came up with, let me know what you guys think!

ChatGPT Conversation: https://chatgpt.com/share/671e8f02-6ba8-8013-9b2e-9febd5987d6f

**Review by leed85 — APPROVED** [review]
